apps/mobile/src/components/TradeChecklistFlow/atoms.ts

```diff
@@ -0,0 +1,164 @@
+  return atom(
+    (get) =>
+      get(availableDateTimesAtom).find(
+        (dateTime) => dateTime.to === timestampTo
+      )?.to ?? unixMillisecondsNow(),
+    (get, set, selectedTimeTo: SetStateAction<UnixMilliseconds>) => {
+      const availableDateTimes = get(availableDateTimesAtom)
+      const selectedDateTime = getValueFromSetStateActionOfAtom(selectedTimeTo)(
+        () =>
+          get(availableDateTimesAtom).find(
+            (dateTime) => dateTime.to === selectedTimeTo
+          )?.to ?? unixMillisecondsNow()
+      )
+
+      const dateTimeToChange = availableDateTimes.find((dateTime) =>
+        DateTime.fromMillis(dateTime.to).hasSame(
+          DateTime.fromMillis(selectedDateTime),
+          'day'
+        )
+      )
+
+      set(
+        availableDateTimesAtom,
+        availableDateTimes.map((dateTime) =>
+          dateTime.to === dateTimeToChange?.to
+            ? {...dateTime, to: selectedDateTime}
+            : dateTime
+        )
+      )
+    }
+  )
+}
+
+export function createTimeOptionAtomForTimeFromDropdown(
+  timestamp: UnixMilliseconds
+): WritableAtom<UnixMilliseconds, [SetStateAction<UnixMilliseconds>], void> {
+  return atom(
+    (get) =>
+      get(availableDateTimesAtom).find(
+        (dateTime) => dateTime.from === timestamp
+      )?.from ?? unixMillisecondsNow(),
+    (get, set, selectedTimeFrom: SetStateAction<UnixMilliseconds>) => {
+      const availableDateTimes = get(availableDateTimesAtom)
+      const selectedDateTime = getValueFromSetStateActionOfAtom(
+        selectedTimeFrom
+      )(
+        () =>
+          get(availableDateTimesAtom).find(
+            (dateTime) => dateTime.from === selectedTimeFrom
+          )?.from ?? unixMillisecondsNow()
+      )
+
+      const dateTimeToChange = availableDateTimes.find((dateTime) =>
+        DateTime.fromMillis(dateTime.from).hasSame(
+          DateTime.fromMillis(selectedDateTime),
+          'day'
+        )
+      )
+
+      set(
+        availableDateTimesAtom,
+        availableDateTimes.map((dateTime) =>
+          dateTime.from === dateTimeToChange?.from
+            ? {
+                ...dateTime,
+                from: selectedDateTime,
+                to:
+                  dateTime.to < selectedDateTime
+                    ? selectedDateTime
+                    : dateTime.to,
+              }
+            : dateTime
+        )
+      )
+    }
+  )
+}
+
+export const removeTimestampFromAvailableAtom = atom(
+  null,
+  (get, set, date: UnixMilliseconds) => {
+    const availableDateTimes = get(availableDateTimesAtom)
+    set(
+      availableDateTimesAtom,
+      availableDateTimes.filter((dateTime) => dateTime.date !== date)
+    )
+  }
+)
+
+export const mainTradeCheckListScreenStateAtom =
+  atom<MainTradeCheckListScreenState>([
+    {
+      type: 'DATE_AND_TIME',
+      status: 'unknown',
+      data: [],
+    },
+  ])
+
+export const createTradeCheckListStateFromChatActionAtom = atom(
```

Comment:
lets make the name more clear
```suggestion
export const syncTradeChecklistWithChatActionAtom = atom(
```